Tasks/AzureRmWebAppDeployment/azurermwebappcontainerdeployment.ts

```diff
@@ -0,0 +1,65 @@
+import tl = require('vsts-task-lib/task');
+import util = require('util');
+
+var azureRESTUtility = require ('azurerest-common/azurerestutility.js');
+var parameterParser = require("./parameterparser.js").parse;
+
+export async function deployWebAppImage(endPoint, resourceGroupName, webAppName) {
+    var startupCommand = tl.getInput('StartupCommand', false);
+    var appSettings = tl.getInput('AppSettings', false);
+    var imageSourceAndTag;
+
+    // Construct the image
+    var dockerNamespace = tl.getInput('DockerNamespace', true);
+    var dockerRepository = tl.getInput('DockerRepository', true);
+    var dockerImageTag = tl.getInput('DockerImageTag', false);
+
+    if(dockerImageTag) {
+        imageSourceAndTag = dockerNamespace + "/" + dockerRepository + ":" + dockerImageTag;          
+    } else {
+        imageSourceAndTag = dockerNamespace + "/" + dockerRepository;
+    }
+
+    if(imageSourceAndTag)
+    {
+        tl.debug("Deploying the image " + imageSourceAndTag + " to the webapp " + webAppName);
+        
+        appSettings = appSettings ? appSettings.trim() : "";        
+        appSettings = "-DOCKER_CUSTOM_IMAGE_NAME " + imageSourceAndTag + " " + appSettings;
+    
+        // Update webapp application setting
+        var webAppSettings = await azureRESTUtility.getWebAppAppSettings(endPoint, webAppName, resourceGroupName, false, null);
```

Comment:
Instead of hard coding the slot param to false, should you then pass the flag we get for slot input as is. Right now it will be false anyway, later if it is supported, you get the benefit for free.